src/main/java/hudson/tasks/junit/CaseResult.java

```diff
@@ -318,19 +317,17 @@ public static CaseResult parse(SuiteResult parent, final XMLStreamReader reader,
                         break;
                     case "properties":
                         r.properties = new HashMap<>();
-                        parseProperties(r.properties, reader, ver);
+                        parseProperties(r.properties, reader, context, ver);
                         break;
                     default:
-                        if (LOGGER.isLoggable(Level.FINEST)) {
-                            LOGGER.finest("CaseResult.parse encountered an unknown field: " + elementName);
-                        }
+                        LOGGER.finest(() -> "Unknown field in " + context + ": " + elementName);
```

Comment:
I also checked whether `XMLReader.getLocation` included a file name to avoid having to pass `context` everywhere, but `getPublicID` and `getSystemID` just return null. We could add line and column numbers to give even more context, but I don't really think it would be helpful in practice.